crates/llama-cpp-bindings/src/lib.rs

```diff
@@ -17,6 +17,20 @@ mod ffi {
         text: String,
     }
 
+    extern "Rust" {
+        type LlamaInitRequest;
+        fn id(&self) -> u32;
+        fn prompt(&self) -> &str;
+        fn max_input_length(&self) -> usize;
+        fn seed(&self) -> u64;
+        fn temperature(&self) -> f32;
+
+        fn check_candidate(&self, candidate: &[u8]) -> bool;
+        fn accept_candidate(&self, candidate: &[u8]);
```

Comment:
So `check_candidate` is likely where we'd want to do be performing the tree-sitter check. Thank you for adding this logic, I get a bit more daunted by C++ code. With these interfaces, it should be pretty easy to add tree-sitter integration.